### PR TITLE
Add namespace to operator log messages

### PR DESCRIPTION
**Pull request by doxiao**
Add domain namespace to operator's log messages.
1) When the current fiber and DomainPresenceInfo are both available, LoggingFormatter uses the existing DomainPresenceInfo to obtain the namespace information.
2) When the current fiber is available but DomainPresenceInfo is not available, add a new component to the packet to pass the namespace to the LoggingFormatter.
3) When the current fiber is not available, add the logging context to the thread local to pass the namespace to the LoggingFormatter. This is not ideal; still looking into possible alternatives.
4) Use ThreadLocal for log messages in the Watch events.

Jenkins job had one known failure in ItSessionMigration.testSessionMigration https://build.weblogick8s.org:8443/view/all/job/weblogic-kubernetes-operator-kind-new/350

### CODE REVIEW

File: operator/src/main/java/oracle/kubernetes/operator/DomainProcessorImpl.java [hunk 1]
```diff
@@ -231,8 +232,23 @@ public void reportSuspendedFibers() {
...
+                    // find out the domainUID 
+                    Packet packet = fiber == null ? null : fiber.getPacket();
+                    String domainUid = null;
+                    if (packet != null) {
+                      DomainPresenceInfo info = packet.getSpi(DomainPresenceInfo.class);
+                      if (info != null) {
+                        domainUid = info.getDomainUid();
+                      }
+                    }
+                    LoggingContext.context(new LoggingContext().namespace(namespace).domainUid(domainUid));
```
Comment (alai8) [inline operator/src/main/java/oracle/kubernetes/operator/DomainProcessorImpl.java:245]: consider adding a utility method in LoggingContext that handles setting the context, invoke the code, and removal of context, something like `LoggingContext.runUnderContext(LoggingContext, Runnable)`.

**Comment by alai8** [comment]
> When the current fiber is not available, add the logging context to the thread local

we need to be careful that while under a logging context, the fiber won't be suspended and the thread is used for running step from another fiber.

**Comment by alai8** [comment]
Not sure if this PR is work in progress, but it would be nice to add some unit tests before merging. Thanks

File: operator/src/main/java/oracle/kubernetes/operator/ServerStatusReader.java [hunk 2]
```diff
@@ -167,6 +168,8 @@ public NextAction apply(Packet packet) {
             String state = null;
             ClientPool helper = ClientPool.getInstance();
             ApiClient client = helper.take();
+            String namespace = pod.getMetadata().getNamespace();
+            LoggingContext.context(new LoggingContext().namespace(namespace));
```
Comment (tbarnes-us) [inline operator/src/main/java/oracle/kubernetes/operator/ServerStatusReader.java:172]: Does this need a corresponding LoggingContext().remove() in a finally?

File: operator/src/main/java/oracle/kubernetes/operator/DomainProcessorImpl.java (hunk 1 above)
Comment (tbarnes-us) [reply operator/src/main/java/oracle/kubernetes/operator/DomainProcessorImpl.java:245]: Or maybe have a Closable wrapper to allow for a 'try (closable) {}' pattern?

Comment (doxiao) [reply operator/src/main/java/oracle/kubernetes/operator/DomainProcessorImpl.java]: Changed to use AutoCloseable.

**Comment by doxiao** [comment]
> Not sure if this PR is work in progress, but it would be nice to add some unit tests before merging. Thanks

Russell added the unit tests for LoggignFormatter to cover what is there without the changes in this PR. I then added test cases for the new code logic.

File: operator/src/main/java/oracle/kubernetes/operator/ServerStatusReader.java (hunk 2 above)
Comment (doxiao) [reply operator/src/main/java/oracle/kubernetes/operator/ServerStatusReader.java]: fixed

**Comment by doxiao** [comment]
The latest code is being tested in the Jenkins job https://build.weblogick8s.org:8443/view/all/job/weblogic-kubernetes-operator-kind-new/408.

**Pull request merged**